src/ProjectTemplates/BlazorWasm.ProjectTemplates/content/BlazorWasm-CSharp/Client/wwwroot/service-worker.published.js

```diff
@@ -0,0 +1,48 @@
+// Caution! Be sure you understand the caveats before publishing an application with
+// offline support. See https://aka.ms/blazor-offline-considerations
+
+self.importScripts('./service-worker-assets.js');
+self.addEventListener('install', event => event.waitUntil(onInstall(event)));
+self.addEventListener('activate', event => event.waitUntil(onActivate(event)));
+self.addEventListener('fetch', event => event.respondWith(onFetch(event)));
+
+const cacheNamePrefix = 'offline-cache-';
+const cacheName = `${cacheNamePrefix}${self.assetsManifest.version}`;
+const offlineAssetsInclude = [ /\.dll$/, /\.pdb$/, /\.wasm/, /\.html/, /\.js$/, /\.json$/, /\.css$/, /\.woff$/, /\.png$/, /\.jpe?g$/, /\.gif$/ ];
+const offlineAssetsExclude = [ /^service-worker\.js$/ ];
+
```

Comment:
You don't need include and exclude patterns. The reason is that anything that is not included will be already excluded. Am I wrong?

Reply:
I would recommend we only do exclude list, as missing something on include breaks your app, but missing something on exclude just makes it more bloated.

Reply:
I'm not sure also if we could do this, but as a way of helping customers identify issues when they add libraries, you could print the list of things that were excluded and the list of things that weren't.

Reply:
Include and exclude are both convenient. In the code here, we include all `.js` files by default, but also exclude `service-worker.js`. It's really inconvenient to express a rule like that without having both include and exclude options.

> I would recommend we only do exclude list,

I'd prefer to have an inclusion list too. Otherwise, totally unanticipated file types e.g., (`LICENSE` with no extension, or `MassiveDocsFile.pdf`) will end up getting fetched cached by default.

> I'm not sure also if we could do this, but as a way of helping customers identify issues when they add libraries, you could print the list of things that were excluded and the list of things that weren't.

I'm not sure where you think this should be printed, but I'd be happy to let developers use the built-in browser dev tools to understand what's getting fetched. They can review the contents of the cache in the "Application Storage" part of the dev tools, and if they accidentally failed to cache something that's needed, they will see it as a 404 in the network tab when testing offline mode.

Reply:
Apparently I don't know set arithmetic, so consider my comment withdrawn.

I'm happy pointing devs to the cache